x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2857 @@
+
+#### Concrete field diff algorithms by type
+
+Depending on the specific field or type of field we might want to apply a specific merging algorithm when conflicts arise. Let's propose different types.
+
+##### String fields
+
+For string fields, in case of three-way conflicts, we will make a best effort to reconcile any modifications done by the user and any updates proposed by Elastic.
+
+Examples: `name`, `description`, `setup`, `note` (Investigation guide)
+
+  <table>
+    <thead>
+      <tr>
+        <th style="border-right:3px solid black">Use case</th>
+        <th>Base version</th>
+        <th>Current version</th>
+        <th style="border-right:3px solid black">Target version</th>
+        <th>Merged version (output)</th>
+      </tr>
+    </thead>
+    <tbody align="center">
+      <tr>
+        <td style="border-right:3px solid black">Keep user customizations when whole string conflicts</td>
+        <td><code>Prebuilt rule name</code></td>
+        <td><code>Completely different string</code></td>
+        <td style="border-right:3px solid black"><code>Updated prebuilt rule name</code></td>
+        <td><code>Completely different string</code></td>
+      </tr>
+      <tr>
+        <td style="border-right:3px solid black">Keep user-added prefixes and accept other modifications</td>
+        <td><code>Prebuilt rule name</code></td>
+        <td><code>[Security Solution] Prebuilt rule name</code></td>
+        <td style="border-right:3px solid black"><code>Updated prebuilt rule name</code></td>
+        <td><code>[Security Solution] Updated prebuilt rule name</code></td>
+      </tr>
+      <tr>
+        <td style="border-right:3px solid black">Keep user-added suffixes and accept other modifications</td>
+        <td><code>Prebuilt rule name</code></td>
+        <td><code>Prebuilt rule name - (Custom version 3)</code></td>
+        <td style="border-right:3px solid black"><code>Updated prebuilt rule name</code></td>
+        <td><code>Updated prebuilt rule name - (Custom version 3)</code></td>
+      </tr>
+      <tr>
+        <td style="border-right:3px solid black">Keep user customizations on conflicts, but add updates in sections with no conflicts</td>
+        <td><code>Prebuilt rule name</code></td>
+        <td><code>Customized prebuilt rule name</code></td>
+        <td style="border-right:3px solid black"><code>Updated prebuilt rule name. Much better.</code></td>
+        <td><code>Customized prebuilt rule name. Much better.</code></td>
+      </tr>
+    </tbody>
+  </table>
+
+
+##### Number fields
+
+Number fields should be treated as a whole unit, i.e. not breakable by digits. Therefore, there is only one possibility of conflicts, the scenario (A B C). In that case, **keep the current version** with the user customization.
```

Comment:
Are we going to make the "if conflict use the current version as the merged version" approach universal to all rule fields and diff algorithms? If so, let's explicitly propose this in the RFC at the beginning of the "Changes to upgrade _review endpoint" section.

Reply:
Explanation added to the text intoduction of section `### Concrete field diff algorithms by type`